crates/flipperzero/src/furi/log/mod.rs

```diff
@@ -0,0 +1,166 @@
+//! Furi Logging system.
+
+mod metadata;
+pub use metadata::{Level, LevelFilter};
+
+/// The standard logging macro.
+///
+/// This macro will generically log with the specified `Level` and C-style format argument
+/// list.
+///
+/// # Examples
+///
+/// ```
+/// use flipperzero::{error, furi::log::Level};
+///
+/// # fn main() {
+/// let error_code = 42;
+/// log!(Level::ERROR, "Failed to handle the florp: %d", error_code);
+/// log!(target: "events", Level::INFO, "Finished the documentation!");
+/// # }
+/// ```
+#[macro_export]
+macro_rules! log {
+    (target: $target:expr, $lvl:expr, $msg:expr $(, $arg:expr)*) => ({
+        if $lvl <= $crate::furi::log::LevelFilter::current() {
```

Comment:
That is indeed currently the case, but that assumes we continue to rely on the C side for `printf` formatting. Once we use `ufmt` and `FuriString` to format the log output before calling `sys::furi_log_print_format`, we will want to avoid performing any of that if we don't have to.